projects/laji/src/app/shared/service/dialog.service.ts

```diff
@@ -20,24 +20,28 @@ export class DialogService {
     return this.createDialog(message, false);
   }
 
-  confirm(message: string, onServer = false): Observable<boolean> {
+  confirm(message: string, confirmLabel?: string, onServer = false): Observable<boolean> {
     if (this.platformService.isServer) {
       return ObservableOf(onServer);
     }
-    return this.createDialog(message);
+    return this.createDialog(message, true, confirmLabel);
   }
 
-  prompt(message: string, _default?: string): Observable<string | null> {
-    return this.createDialog(message, true, true, _default);
+  prompt(message: string, confirmLabel?: string, _default?: string): Observable<string | null> {
+    return this.createDialog(message, true, confirmLabel, true, _default);
   }
 
-  private createDialog(message: string, showCancel?: boolean): Observable<boolean>;
-  private createDialog(message: string, showCancel: boolean, prompt: true, promptDefault?: string): Observable<string | null>;
-  private createDialog(message: string, showCancel = true, prompt = false, promptDefault?: string): Observable<boolean | string | null> {
+  private createDialog(message: string, showCancel?: boolean, confirmLabel?: string): Observable<boolean>;
+  private createDialog(message: string, showCancel: boolean, confirmLabel: string | undefined, prompt: true, promptDefault?: string): Observable<string | null>;
+  private createDialog(message: string, showCancel = true, confirmLabel?: string, prompt = false, promptDefault?: string): Observable<boolean | string | null> {
+      const initialState = {message, prompt, promptValue: promptDefault ?? ''} as any;
```

Comment:
> If I go to trip form, input geometry (and nothing else) and click "save draft", I get the "Sorry, something went wrong..." message. Is that supposed to happen?

Yes, that's intentional. The form is aware only of schema + custom validations, if you look at the document save response it says: `gatheringEvent: {dateBegin: {errors: ["Date is required"]}}`. We could improve cases like this by adding stricter validation to the forms (e.g. making date begin a required field, and then trying to save as draft would display a proper validation error, because json schema validators are run against draft save). However, the currently displayed error modal is for generic errors.

> This function is starting to be quite hard to read, could it be made simpler by using an object (e.g. private createDialog(config: IDialogConfig) {})

Good idea, I'll get to it.